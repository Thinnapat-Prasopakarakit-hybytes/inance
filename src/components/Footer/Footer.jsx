import React from "react";
import "./Footer.scss";
const Footer = () => {
  return (
    <footer class="footer_section">
      <div class="container">
        <p>
          &copy; <span>{new Date().getFullYear()}</span> All Rights Reserved By
          <a href="https://html.design/">Free Html Templates</a>
        </p>
      </div>
    </footer>
  );
};

export default Footer;
