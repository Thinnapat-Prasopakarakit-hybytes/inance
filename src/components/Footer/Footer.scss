@use "../../styles/partials/variables" as *;
@use "../../styles/partials/mixins" as *;

.footer_section {
  background-color: #0a0f43;

  p {
    margin: 0;
    padding: 25px 0;
    color: $white;
    text-align: center;
    border-top: 0.8px solid $white;
  }

  a {
    color: inherit;
  }
}
