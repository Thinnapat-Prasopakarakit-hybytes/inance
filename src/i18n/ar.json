{
  "header": {
    "phone": "اتصل : +01 123455678990",
    "email": "البريد الإلكتروني : demo@gmail.com"
  },
  "nav": {
    "home": "الرئيسية",
    "about": "من نحن",
    "services": "خدماتنا",
    "contact": "اتصل بنا"
  },
  "feature": {
    "repair": "إصلاح",
    "improve": "تحسين",
    "maintain": "صيانة"
  },
  "slider": {
    "firstLine": "إصلاح و",
    "secondLine": "صيانة",
    "thirdLine": "خدمات",
    "description": "لوريم إيبسوم دولور سيت أميت، كونسيكتيتور أديباسينغ إليت. نيمو إنيم، فيريتات أوديو، ليبريتاس لوريم إيبسوم إنتيجريت. مينيما كوي أوت أتشينت إينيم أوديو أكوربوز، سوسيب تينيت أورناريه إلتراتو نيمو ريسيتات. إني أمبيد كوي أتالوغ إيرابيل، إيس أونجتوت أوديو لوريم. أوديا فان إينيس ديوس غريلانديسكو فوريس، أرتيموس كانايور.",
    "contact": "اتصل بنا"
  },
  "about": {
    "title": "من نحن",
    "readMore": "اقرأ المزيد",
    "description": "هناك حقيقة مثبتة منذ زمن طويل وهي أن المحتوى المقروء لصفحة ما سيلهي القارئ عن التركيز على الشكل الخارجي للنص."
  },
  "professional": {
    "title": "نحن نقدم خدمات",
    "subtitle": "منزلية احترافية",
    "description": "هناك حقيقة مثبتة منذ زمن طويل وهي أن المحتوى المقروء لصفحة ما سيلهي القارئ عن التركيز على الشكل الخارجي للنص.",
    "readMore": "اقرأ المزيد"
  },
  "services": {
    "title": "خدماتنا",
    "maintenance": "صيانة",
    "electrical": "كهرباء",
    "plumbing": "سباكة",
    "description": "هناك حقيقة مثبتة منذ زمن طويل وهي أن المحتوى المقروء لصفحة ما سيلهي القارئ",
    "viewMore": "عرض المزيد"
  },
  "contact": {
    "title": "اتصل بنا",
    "name": "الاسم",
    "phone": "رقم الهاتف",
    "email": "البريد الإلكتروني",
    "message": "الرسالة",
    "send": "إرسال"
  },
  "client": {
    "title": "ماذا يقول عملاؤنا",
    "name": "جورش موريك",
    "testimonial": "يقوم بتقسيم النص إلى أجزاء حسب الضرورة، مما يجعله أول مولد حقيقي على الإنترنت. يستخدم قاموسًا يحتوي على أكثر من 200 كلمة لاتينية، مع عدد قليل من تراكيب الجمل النموذجية، لتوليد نص لوريم إيبسوم."
  },
  "info": {
    "getInTouch": "تواصل معنا",
    "text": "لوريم إيبسوم هو نص تجريدي بسيط",
    "followUs": "تابعنا"
  },
  "footer": {
    "copyright": "جميع الحقوق محفوظة لـ",
    "company": "قوالب HTML المجانية"
  },
  "notFound": {
    "text": "الصفحة غير موجودة",
    "description": "عذراً! الصفحة التي تبحث عنها غير موجودة",
    "link": "العودة إلى الصفحة الرئيسية"
  }
}
