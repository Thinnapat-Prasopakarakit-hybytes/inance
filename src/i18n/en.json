{
  "header": {
    "phone": "Call : +01 123455678990",
    "email": "Email : demo@gmail.com"
  },
  "nav": {
    "home": "Home",
    "about": "About",
    "services": "Services",
    "contact": "Contact Us"
  },
  "feature": {
    "repair": "Repair",
    "improve": "Improve",
    "maintain": "Maintain"
  },
  "slider": {
    "firstLine": "Repair and",
    "secondLine": "Maintenance",
    "thirdLine": "Services",
    "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Qui harum voluptatem adipisci. Quos molestiae saepe dicta nobis pariatur, tempora iusto, ad possimus soluta hic praesentium mollitia consequatur beatae, aspernatur culpa.",
    "contact": "Contact Us"
  },
  "about": {
    "title": "About us",
    "readMore": "Read More",
    "description": "There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected humour, or randomised words which don't look even slightly believable."
  },
  "professional": {
    "title": "We Provide Professional",
    "subtitle": "Home Services.",
    "description": "randomised words which don't look even slightly believable. If you are going to use a passage of Lorem Ipsum, you need to be sure there isn't anything embarrassing hidden in the middle of text. All randomised words which don't look even slightly",
    "readMore": "Read More"
  },
  "services": {
    "title": "Our Services",
    "maintenance": "Maintenance",
    "electrical": "Electrical",
    "plumbing": "Plumbing",
    "description": "when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal",
    "viewMore": "View More"
  },
  "contact": {
    "title": "Contact Us",
    "name": "Name",
    "phone": "Phone Number",
    "email": "Email",
    "message": "Message",
    "send": "SEND"
  },
  "client": {
    "title": "What Our Clients Say",
    "name": "Jorch morik",
    "testimonial": "chunks as necessary, making this the first true generator on the Internet. It uses a dictionary of over 200 Latin words, combined with a handful of model sentence structures, to generate Lorem Ipsum"
  },
  "info": {
    "getInTouch": "Get In Touch",
    "text": "Lorem Ipsum is simply dummy text",
    "followUs": "Follow Us"
  },
  "footer": {
    "copyright": "All Rights Reserved By",
    "company": "Free Html Templates"
  },
  "notFound": {
    "text": "Page not found",
    "description": "Oops! The page you are looking for does not exist",
    "link": "Back to Home"
  }
}
