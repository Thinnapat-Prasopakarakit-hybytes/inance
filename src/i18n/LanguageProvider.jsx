import { useState, createContext, useEffect } from "react";
import { IntlProvider } from "react-intl";
import { useLocation, useParams } from "react-router-dom";
import en from "./en.json";
import ar from "./ar.json";
import { HelmetProvider } from "react-helmet-async";
import SEO from "../components/SEO";
import { logEvent, logLanguageChange } from "../analytics";

export const LanguageContext = createContext();

const messages = {
  en,
  ar,
};

const LanguageProvider = ({ children }) => {
  const { lang } = useParams();
  const { pathname } = useLocation();
  const [locale, setLocale] = useState(lang === "ar" ? "ar" : "en");

  useEffect(() => {
    if (lang && (lang === "en" || lang === "ar")) {
      localStorage.setItem("lang", lang);
      setLocale(lang);
      document.documentElement.dir = lang === "ar" ? "rtl" : "ltr";
      document.documentElement.lang = lang;
    }
  }, [lang]);

  const handleLanguageChange = (newLocale) => {
    if (locale !== newLocale) {
      logEvent("Language", "Change", newLocale);
      const pathWithoutLang = pathname.replace(/^\/(en|ar)/, "");
      const newPath =
        pathWithoutLang === "" || pathWithoutLang === "/"
          ? `/${newLocale}`
          : `/${newLocale}${pathWithoutLang}`;

      window.location.href = newPath;
    }
  };

  return (
    <HelmetProvider>
      <LanguageContext.Provider value={{ locale, handleLanguageChange }}>
        <IntlProvider
          messages={messages[locale]}
          locale={locale}
          defaultLocale="en"
        >
          <SEO lang={locale} />
          {children}
        </IntlProvider>
      </LanguageContext.Provider>
    </HelmetProvider>
  );
};

export default LanguageProvider;
