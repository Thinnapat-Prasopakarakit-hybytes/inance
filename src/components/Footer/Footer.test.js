import React from "react";
import { render, screen } from "@testing-library/react";
import Footer from "./Footer";
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";
import { LanguageContext } from "../../i18n/LanguageProvider";
import { IntlProvider } from "react-intl";
import en from "../../i18n/en.json";
import ar from "../../i18n/ar.json";

const messages = {
  en,
  ar,
};

const renderFooter = (locale = "en") => {
  return render(
    <BrowserRouter>
      <LanguageContext.Provider value={{ locale }}>
        <IntlProvider messages={messages[locale]} locale={locale}>
          <Footer />
        </IntlProvider>
      </LanguageContext.Provider>
    </BrowserRouter>
  );
};

describe("Footer Component", () => {
  it("should render copyright text", () => {
    renderFooter();
    expect(screen.getByText(/All Rights Reserved By/i)).toBeInTheDocument();
  });

  it("should render current year", () => {
    renderFooter();
    const currentYear = new Date().getFullYear().toString();
    expect(screen.getByText(currentYear)).toBeInTheDocument();
  });

  it("should render credit link", () => {
    renderFooter();
    const link = screen.getByRole("link", { name: /Free Html Templates/i });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", "https://html.design/");
  });

  it("should match snapshot", () => {
    const { asFragment } = renderFooter();
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Footer Component - Arabic", () => {
  it("should render Arabic copyright text", () => {
    renderFooter("ar");
    expect(screen.getByText(/جميع الحقوق محفوظة/i)).toBeInTheDocument();
  });

  it("should render company name in Arabic", () => {
    renderFooter("ar");
    expect(screen.getByText(/قوالب HTML المجانية/i)).toBeInTheDocument();
  });

  it("should maintain link functionality in Arabic", () => {
    renderFooter("ar");
    const link = screen.getByRole("link");
    expect(link).toHaveAttribute("href", "https://html.design/");
  });

  it("should match Arabic snapshot", () => {
    const { asFragment } = renderFooter("ar");
    expect(asFragment()).toMatchSnapshot();
  });
});
