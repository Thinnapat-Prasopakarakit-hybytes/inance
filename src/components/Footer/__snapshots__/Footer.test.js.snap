// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer_section"
  >
    <div
      class="container"
    >
      <p>
        Â© 
        <span>
          2025
        </span>
         
        <a
          href="https://html.design/"
        >
          All Rights Reserved By Free Html Templates
        </a>
      </p>
    </div>
  </footer>
</DocumentFragment>
`;
